src/main/java/com/microsoft/sqlserver/jdbc/SecureStringUtil.java

```diff
@@ -127,13 +128,19 @@ byte[] getEncryptedBytes(char[] chars) throws SQLServerException {
     char[] getDecryptedChars(byte[] bytes) throws SQLServerException {
         if (bytes == null)
             return null;
+
+        byte[] iv = new byte[IV_LENGTH];
+        byte[] encryptedBytes = new byte[bytes.length - IV_LENGTH];
+        System.arraycopy(bytes, 0, iv, 0, IV_LENGTH);
+        System.arraycopy(bytes, IV_LENGTH, encryptedBytes, 0, bytes.length - IV_LENGTH);
+
         GCMParameterSpec ivParamSpec = new GCMParameterSpec(TAG_LENGTH * 8, iv);
 
         byte[] plainText = null;
         try {
             decryptCipher.init(Cipher.DECRYPT_MODE, secretKey, ivParamSpec);
 
-            plainText = decryptCipher.doFinal(bytes);
+            plainText = decryptCipher.doFinal(encryptedBytes);

```

Comment:
You can avoid array copy with `decryptCipher.doFinal(bytes, IV_LENGTH, bytes.length - IV_LENGTH);`